File: wanjicardc-ios/Documents/鏂囨。/锛堟槗鑱旀敮浠橈級11_鏍囧噯SDK鐗坃20160113/鏈嶅姟鍣ㄦ帴鍙emo/phpDemo/src/com/merchant/demo/DemoTest.php
<?php

/**
 * 各版本通用接口测试demo,分别对以下接口进行测试
 * 1、订单查询接口（OrderQueryTest）
 * 2、订单冲正接口（OrderReverseTest）
 * 3、订单退款申请接口（OrderRefundReqTest）
 * 4、订单退款查询接口（OrderRefundQueryTest）
 * 5、互联网金融行业银行卡解除绑定接口（UnboundBankCardTest）
 */

require_once './src/com/merchant/demo/Constants.php';
require_once './src/com/payeco/tools/Xml.php';
require_once './src/com/payeco/client/TransactionClient.php';

class DemoTest {
	/**
	 * 订单查询接口测试
	 */
	static function OrderQueryTest($orderId) {
		// 设置参数
		$merchantId = Constants::getMerchantId();
		$merchOrderId = $orderId;   //需要填写已经存在的商户订单号
		$tradeTime = Tools::getSysTime();

		// 调用查询接口
		Log::logFile("-------订单查询接口测试-------------------------");
		Log::logFile("orderId=".$orderId);
		try {
			$retXml = new Xml();
			//接口参数请参考TransactionClient的参数说明
			$ret = TransactionClient::OrderQuery($merchantId, $merchOrderId, $tradeTime,
					Constants::getMerchantRsaPrivateKey(), Constants::getPayecoRsaPublicKey(), 
					Constants::getPayecoUrl(), $retXml);
			if(strcmp("0000", $ret)){
				Log::logFile("商户下单接口测试失败！：retCode=".$ret."; msg=".$retXml->getRetMsg());
				return;
			}
		} catch (Exception $e) {
			Log::logFile("订单查询接口测试失败！：".$e->getMessage());
			return;
		}
		Log::logFile("订单查询接口测试----ok");
		Log::logFile("------------------------------------------------");
	}
	
	/**
	 * 订单冲正接口测试
	 */
	static function OrderReverseTest($orderId, $amt) {
		// 设置参数
		$amount = $amt;
		$merchantId = Constants::getMerchantId();
		$merchOrderId = $orderId;  //需要填写已经存在的商户订单号
		$tradeTime = Tools::getSysTime();

		// 调用冲正接口
		Log::logFile("-------订单冲正接口测试-------------------------");
		Log::logFile("orderId=".$orderId);
		try {
			$retXml = new Xml();
			//接口参数请参考TransactionClient的参数说明
			$ret = TransactionClient::OrderReverse($merchantId, $merchOrderId, $amount, $tradeTime, 
					Constants::getMerchantRsaPrivateKey(), Constants::getPayecoRsaPublicKey(),
					Constants::getPayecoUrl(), $retXml);
			if(strcmp("0000", $ret)){
				Log::logFile("商户下单接口测试失败！：retCode=".$ret."; msg=".$retXml->getRetMsg());
				return;
			}
		} catch (Exception $e) {
			Log::logFile("订单冲正接口测试失败！：".$e->getMessage());
			return;
		}
		Log::logFile("订单冲正接口测试----ok");
		Log::logFile("------------------------------------------------");
	}
	
	/**
	 * 订单退款申请接口测试
	 */
	static function OrderRefundReqTest($orderId, $amt) {
		// 设置参数
		$amount = $amt;
		$merchantId = Constants::getMerchantId();
		$merchOrderId = $orderId;  //需要填写已经存在的商户订单号
		$merchRefundId = Tools::currentTimeMillis(); // 退款申请号
		$tradeTime = Tools::getSysTime();
	
		// 调用冲正接口
		Log::logFile("-------订单退款申请接口测试-------------------------");
		try {
			$retXml = new Xml();
			
			//接口参数请参考TransactionClient的参数说明
			$ret = TransactionClient::OrderRefundReq($merchantId, $merchOrderId, $merchRefundId, $amount, $tradeTime,
							Constants::getMerchantRsaPrivateKey(), Constants::getPayecoRsaPublicKey(), 
							Constants::getPayecoUrl(), $retXml);
			if(strcmp("0000", $ret)){
				Log::logFile("订单退款申请接口测试失败！：retCode=".$ret."; msg=".$retXml->getRetMsg());
				return;
			}
		} catch (Exception $e) {
			Log::logFile("订单退款申请接口测试失败！：".$e->getMessage());
			return;
		}
		Log::logFile("订单退款申请接口测试----ok");
		Log::logFile("------------------------------------------------");
	}
	
	/**
	 * 订单退款结果查询接口测试
	 */
	static function OrderRefundQueryTest($orderId, $refundId) {
		// 设置参数
		$merchantId = Constants::getMerchantId();
		$merchOrderId = $orderId;  //需要填写已经存在的商户订单号
		$merchRefundId = $refundId; // 退款申请号
		$tradeTime = Tools::getSysTime();
	
		// 调用冲正接口
		Log::logFile("-------订单退款结果查询接口测试-------------------------");
		try {
			$retXml = new Xml();
			//接口参数请参考TransactionClient的参数说明
			$ret = TransactionClient::OrderRefundQuery($merchantId, $merchOrderId, $merchRefundId, $tradeTime,
					Constants::getMerchantRsaPrivateKey(), Constants::getPayecoRsaPublicKey(), 
					Constants::getPayecoUrl(), $retXml);
			if(strcmp("0000", $ret)){
				Log::logFile("订单退款结果查询接口测试失败！：retCode=".$ret."; msg=".$retXml->getRetMsg());
				return;
			}
		} catch (Exception $e) {
			Log::logFile("订单退款结果查询接口测试失败！：".$e->getMessage());
			return;
		}
		Log::logFile("订单退款结果查询接口测试----ok");
		Log::logFile("------------------------------------------------");
	}
	
	////==================以下接口为特殊行业或定制化的接口========================
	/**
	 * 互联网金融行业银行卡解除绑定接口测试
	 */
	static function UnboundBankCardTest($bankAccNo) {
		// 设置参数
		$merchantId = Constants::getMerchantId();
		$tradeTime = Tools::getSysTime();
	
		// 调用查询接口
		Log::logFile("-------互联网金融行业银行卡解除绑定接口测试-------------------------");
		try {
			$retXml = new Xml();
			//接口参数请参考TransactionClient的参数说明
			$ret = TransactionClient::UnboundBankCard($merchantId, $bankAccNo, $tradeTime,
					Constants::getMerchantRsaPrivateKey(), Constants::getPayecoRsaPublicKey(), 
					Constants::getPayecoUrl(), $retXml);
			if(strcmp("0000", $ret)){
				Log::logFile("互联网金融行业银行卡解除绑定接口测试失败！：retCode=".$ret."; msg=".$retXml->getRetMsg());
				return;
			}
			Log::logFile("Status=".$retXml->getStatus());
		} catch (Exception $e) {
			Log::logFile("互联网金融行业银行卡解除绑定接口测试失败！：".$e->getMessage());
			return;
		}
		Log::logFile("互联网金融行业银行卡解除绑定接口测试----ok;");
		Log::logFile("------------------------------------------------");
	}
	
	static function Test() {
		//设置打印日志
		Log::setLogFlag(true);
		
		$orderId = "1445935379582"; //填写已经存在成功下单的商户订单号
		DemoTest::OrderQueryTest($orderId);
		echo "订单查询接口调用完成，请到日志文件查询调用结果";
		echo "<br/>";
		
		// 订单冲正接口测试
		$orderId = "1445935379582"; //填写已经存在成功下单的商户订单号，只有支付成功才能冲正
		$amount = "25";	   //冲正额金额需要和订单金额一样
// 		DemoTest::OrderReverseTest($orderId, $amount);
// 		echo "订单冲正接口调用完成，请到日志文件查询调用结果";
// 		echo "<br/>";
		
		// 订单退款申请接口测试
// 		$orderId = "1446610513853"; //填写已经存在成功下单的商户订单号，只有支付成功才能退款申请
// 		$amount = "1";	   //退款的金额不能大于订单金额
// 		DemoTest::OrderRefundReqTest($orderId, $amount);
// 		echo "订单退款申请接口调用完成，请到日志文件查询调用结果";
// 		echo "<br/>";
		
		// 订单退款结果查询接口测试
// 		$orderId = "1446610513853"; //填写已经存在成功下单的商户订单号，只有支付成功才能退款申请
// 		$refundId = "1446718774394";      //退款申请号，需要填写退款申请时提交的申请号，退款结果需要在申请的隔天后才能查询到结果
// 		DemoTest::OrderRefundQueryTest($orderId, $refundId);
// 		echo "订单退款结果查询接口调用完成，请到日志文件查询调用结果";
// 		echo "<br/>";
		
		//互联网金融行业银行卡解除绑定接口测试
// 		$bankAccNo="6227003324126059872";
// 		DemoTest::UnboundBankCardTest($bankAccNo);		
// 		echo "互联网金融行业银行卡解除绑定接口调用完成，请到日志文件查询调用结果";
// 		echo "<br/>";
	}
}
?>

File: wanjicardc-ios/Documents/鏂囨。/锛堟槗鑱旀敮浠橈級11_鏍囧噯SDK鐗坃20160113/鏈嶅姟鍣ㄦ帴鍙emo/JavaDemo/src/com/merchant/demo/DemoTest.java
package com.merchant.demo;

import com.payeco.client.TransactionClient;
import com.payeco.tools.Log;
import com.payeco.tools.Tools;
import com.payeco.tools.Xml;

/**
 * 各版本通用接口测试demo,分别对以下接口进行测试
 * 1、订单查询接口（OrderQueryTest）
 * 2、订单冲正接口（OrderReverseTest）
 * 3、订单退款申请接口（OrderRefundReqTest）
 * 4、订单退款查询接口（OrderRefundQueryTest）
 * 5、互联网金融行业银行卡解除绑定接口（UnboundBankCardTest）
 */
public class DemoTest {
	/**
	 * 订单查询接口测试
	 */
	public static void OrderQueryTest(String orderId) {
		// 设置参数
		String merchantId = Constants.MERCHANT_ID;
		String merchOrderId = orderId;  
		String tradeTime = Tools.getSysTime();

		// 调用查询接口
		System.out.println("-------订单查询接口测试-------------------------");
		try {
			Xml retXml = new Xml();
			//接口参数请参考TransactionClient的参数说明
			String ret = TransactionClient.OrderQuery(merchantId, merchOrderId, tradeTime,
					Constants.MERCHANT_RSA_PRIVATE_KEY, Constants.PAYECO_RSA_PUBLIC_KEY, Constants.PAYECO_URL, retXml);
			if(!"0000".equals(ret)){
				System.out.println("订单查询接口测试失败！：retCode="+ret+"; msg="+retXml.getRetMsg());
				return;
			}
		} catch (Exception e) {
			System.out.println("订单查询接口测试失败！：");
			e.printStackTrace();
			return;
		}
		System.out.println("订单查询接口测试----ok");
		System.out.println("------------------------------------------------");
	}
	
	/**
	 * 订单冲正接口测试
	 */
	public static void OrderReverseTest(String orderId, String amt) {
		// 设置参数
		String amount = amt;
		String merchantId = Constants.MERCHANT_ID;
		String merchOrderId = orderId;  //需要填写已经存在的商户订单号
		String tradeTime = Tools.getSysTime();

		// 调用冲正接口
		System.out.println("-------订单冲正接口测试-------------------------");
		try {
			Xml retXml = new Xml();
			//接口参数请参考TransactionClient的参数说明
			String ret = TransactionClient.OrderReverse(merchantId, merchOrderId, amount, tradeTime, 
					Constants.MERCHANT_RSA_PRIVATE_KEY, Constants.PAYECO_RSA_PUBLIC_KEY, Constants.PAYECO_URL, retXml);
			if(!"0000".equals(ret)){
				System.out.println("订单冲正接口测试失败！：retCode="+ret+"; msg="+retXml.getRetMsg());
				return;
			}
		} catch (Exception e) {
			System.out.println("订单冲正接口测试失败！：");
			e.printStackTrace();
			return;
		}
		System.out.println("订单冲正接口测试----ok");
		System.out.println("------------------------------------------------");
	}

	/**
	 * 订单退款申请接口测试
	 */
	public static void OrderRefundReqTest(String orderId, String amt) {
		// 设置参数
		String amount = amt;
		String merchantId = Constants.MERCHANT_ID;
		String merchOrderId = orderId;  //需要填写已经存在的商户订单号
		String merchRefundId = "" + System.currentTimeMillis(); // 退款申请号
		String tradeTime = Tools.getSysTime();

		// 调用冲正接口
		System.out.println("-------订单退款申请接口测试-------------------------");
		try {
			Xml retXml = new Xml();
			//接口参数请参考TransactionClient的参数说明
			String ret = TransactionClient.OrderRefundReq(merchantId, merchOrderId, merchRefundId, amount, tradeTime, 
					Constants.MERCHANT_RSA_PRIVATE_KEY, Constants.PAYECO_RSA_PUBLIC_KEY, Constants.PAYECO_URL, retXml);
			if(!"0000".equals(ret)){
				System.out.println("订单退款申请接口测试失败！：retCode="+ret+"; msg="+retXml.getRetMsg());
				return;
			}
		} catch (Exception e) {
			System.out.println("订单退款申请接口测试失败！：");
			e.printStackTrace();
			return;
		}
		System.out.println("订单退款申请接口测试----ok");
		System.out.println("------------------------------------------------");
	}	

	/**
	 * 订单退款结果查询接口测试
	 */
	public static void OrderRefundQueryTest(String orderId, String refundId) {
		// 设置参数
		String merchantId = Constants.MERCHANT_ID;
		String merchOrderId = orderId;  //需要填写已经存在的商户订单号
		String merchRefundId = refundId; // 退款申请号
		String tradeTime = Tools.getSysTime();

		// 调用冲正接口
		System.out.println("-------订单退款结果查询接口测试-------------------------");
		try {
			Xml retXml = new Xml();
			//接口参数请参考TransactionClient的参数说明
			String ret = TransactionClient.OrderRefundQuery(merchantId, merchOrderId, merchRefundId, tradeTime, 
					Constants.MERCHANT_RSA_PRIVATE_KEY, Constants.PAYECO_RSA_PUBLIC_KEY, Constants.PAYECO_URL, retXml);
			if(!"0000".equals(ret)){
				System.out.println("订单退款结果查询接口测试失败！：retCode="+ret+"; msg="+retXml.getRetMsg());
				return;
			}
		} catch (Exception e) {
			System.out.println("订单退款结果查询接口测试失败！：");
			e.printStackTrace();
			return;
		}
		System.out.println("订单退款结果查询接口测试----ok");
		System.out.println("------------------------------------------------");
	}	
	
	////==================以下接口为特殊行业或定制化的接口========================
	/**
	 * 互联网金融行业银行卡解除绑定接口测试
	 */
	public static void UnboundBankCardTest(String bankAccNo) {
		// 设置参数
		String merchantId = Constants.MERCHANT_ID;
		String tradeTime = Tools.getSysTime();

		// 调用查询接口
		System.out.println("-------互联网金融行业银行卡解除绑定接口测试-------------------------");
		try {
			Xml retXml = new Xml();
			//接口参数请参考TransactionClient的参数说明
			String ret = TransactionClient.UnboundBankCard(merchantId, bankAccNo, tradeTime,
					Constants.MERCHANT_RSA_PRIVATE_KEY, Constants.PAYECO_RSA_PUBLIC_KEY, Constants.PAYECO_URL, retXml);
			if(!"0000".equals(ret)){
				System.out.println("互联网金融行业银行卡解除绑定接口测试失败！：retCode="+ret+"; msg="+retXml.getRetMsg());
				return;
			}
			System.out.println("Status="+retXml.getStatus());
		} catch (Exception e) {
			System.out.println("互联网金融行业银行卡解除绑定接口测试失败！：");
			e.printStackTrace();
			return;
		}
		System.out.println("互联网金融行业银行卡解除绑定接口测试----ok;");
		System.out.println("------------------------------------------------");
	}
	
	public static void main(String[] args) {
		//设置打印日志
		Log.setLogFlag(true);
		
		// 订单查询接口测试
		String orderId = "1445935379582"; //填写已经存在成功下单的商户订单号
//		OrderQueryTest(orderId);
		
		// 订单冲正接口测试
		orderId = "1445935379582"; //填写已经存在成功下单的商户订单号，只有当天支付成功才能冲正
		String amount = "25";	   //冲正额金额需要和订单金额一样
//		OrderReverseTest(orderId, amount);

		// 订单退款申请接口测试
		orderId = "1445935379582"; //填写已经存在成功下单的商户订单号，只有支付成功才能退款申请
		amount = "1";	   //退款的金额不能大于订单金额
//		OrderRefundReqTest(orderId, amount);
		
		// 订单退款结果查询接口测试
		orderId = "1445935379582"; //填写已经存在成功下单的商户订单号，只有支付成功才能退款申请
		String refundId = "1446005809814";      //退款申请号，需要填写退款申请时提交的申请号，退款结果需要在申请的隔天后才能查询到结果
//		OrderRefundQueryTest(orderId, refundId);
		
		
		
		//互联网金融行业银行卡解除绑定接口测试
		String bankAccNo="6227003324126059872";
		UnboundBankCardTest(bankAccNo);
	}
}
